org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/LimitExceededWarnings.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.client;
+
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.lemminx.customservice.ActionableNotification;
+import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+
+/**
+ * LimitExceededWarnings sends a warning to the
+ * client via a jsonrpc notification
+ */
+public class LimitExceededWarnings {
+
+	/**
+	 * Default instance.
+	 */
+	private static final LimitExceededWarnings DEFAULT_INSTANCE = new LimitExceededWarnings();
+
+	private final Set<String /* file URI */> cache;
+
+	public LimitExceededWarnings() {
```

Comment:
I will not use singleton. I think it should be better that this class should be instanciated in teh XMLTextDocumentService, because you have 2 parameters which change never XMLLanguageClientAPI , and SharedSettings.

 
In other words I will have this constructor to store the 2 final variables client and sharedSettings.

```java
public LimitExceededWarnings(XMLLanguageClientAPI client, SharedSettings sharedSettings) {
   this.client = client;
   this.sharedSettings = sharedSettings;
}
```

Reply:
I've run into a problem, could I pass in `XMLLanguageServer` instead of `XMLLanguageClientAPI`?  Because when `XMLTextDocumentService` is first instantiated, the client is `null`, therefore it calls `new LimitExceededWarnings(null, sharedSettings)`. It is `null` because the client is set after instantiation here:
https://github.com/eclipse/lemminx/blob/c2f5dfd113d71a8c30055c2489843befa7a3fc7f/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLServerLauncher.java#L72

Reply:
nevermind that comment, I think its better to instantiate LimitExceededWarnings (if needed) in
`XMLTextDocumentService#getLimitExceededWarnings`

Reply:
Sorry I had not finished :)

You must define a getLimitExceededWarnings() in the XMLTextDocumentService which checks the null of LimitExceededWarnings and in this case you create the instance of LimitExceededWarnings. At this moment the language client should not be null

Reply:
ok understood, thank you!